common/gpu/intel/meteor-lake/default.nix

```diff
@@ -0,0 +1,17 @@
+{pkgs, lib, config, ...}: let inherit(lib) mkDefault mkIf; in {
+  imports = [ ../. ];
+
+	# only needed to support the i915 driver, can be found using `nix-shell -p pciutils --run "lspci -nn" | grep -oP "VGA.*:\K[0-9a-f]{4}"`
+	options.hardware.intelgpu.deviceID = lib.mkOption { description = "Intel GPU to probe"; };
+
+	config = {
+		# i915 is buggy on meteor lake, xe should be the default
+		hardware.intelgpu.driver = mkDefault "xe";
```

Comment:
I have a Dell Latitude 7450 with an Intel Core Ultra 7 165U.
In my tests so far the Xe driver, while providing the best experience right at boot (correct resolutions etc.), was very unstable.
Everytime I watched a video either with MPV or in the browser the device crashed after max. 10 minutes.
I'm currently running the i915 driver and so far it has been working fine.

Therefore I'm not sure it is fair to say that i915 is buggy on Meteor Lake. However I'm not the best when it comes to hardware.

Reply:
I'm currently using the Dell Precision 5390 with an Intel Core Ultra 7 165H  with this patch and also the Xe driver and haven't noticed any issues so far. But I'm using this system only for 2 days.

Reply:
Hm interesting, that is a very similar CPU.
My setup is here, AFAIK I’m not doing something exotic but maybe miss something obvious.
https://git.2li.ch/Nebucatnetzer/nixos/src/branch/new-notebook/systems/capricorn/default.nix

I tried it again yesterday and Xe failed reproducible when watching a video with the same config except for the `force_probe`.
Which looked like this: https://git.2li.ch/Nebucatnetzer/nixos/commit/edeac6d6922c29e78126386c563e976e79cff23c

@Sh3Rm4n is suspend working for you? If so are you using an encrypted drive?

Reply:
I'm using an encrypted drive and haven't explicitly checked suspend so far, but I think it is currently not working correctly. I'll recheck again.

Reply:
That would be great. My machine just hangs completely after waking up, both with suspend and hibernation and I think I have the partition setup correct.

Reply:
I checked today after setting my laptop to suspend mode yesterday and it correctly went to sleep as far as `journalctl` can tell: 

```
Jan 23 13:39:47 hostname systemd-sleep[22569]: Successfully thawed unit 'user.slice'.
Jan 23 16:59:09 hostname systemd-sleep[231639]: Successfully froze unit 'user.slice'.
Jan 23 16:59:09 hostname systemd-sleep[231639]: Performing sleep operation 'suspend'...
Jan 24 08:22:08 hostname systemd-sleep[231639]: System returned from sleep operation 'suspend'.
Jan 24 08:22:08 hostname systemd-sleep[231639]: Successfully thawed unit 'user.slice'.
```

Reply:
I just see this, the boot is because I killed the system.

```
Jan 22 20:56:19 capricorn systemd[1]: Starting System Suspend...
Jan 22 20:56:19 capricorn systemd-sleep[14013]: Successfully froze unit 'user.slice'.
Jan 22 20:56:19 capricorn systemd-sleep[14013]: Performing sleep operation 'suspend'...
Jan 22 20:56:19 capricorn kernel: PM: suspend entry (s2idle)
Jan 22 20:56:19 capricorn kernel: Filesystems sync: 0.009 seconds
-- Boot 412e6ed5b57b47d3a8552a5e9d53befb --
```

Reply:
> I have a Dell Latitude 7450 with an Intel Core Ultra 7 165U. In my tests so far the Xe driver, while providing the best experience right at boot (correct resolutions etc.), was very unstable. Every time I watched a video either with MPV or in the browser the device crashed after max. 10 minutes. I'm currently running the i915 driver and so far it has been working fine.
> 
> Therefore I'm not sure it is fair to say that i915 is buggy on Meteor Lake. However I'm not the best when it comes to hardware.

Just to be sure I tested this PR and the settings from nixos-hardware explicitly and I'm seeing the same behaviour.
In addition I have to `force_probe` the Xe driver otherwise the notebook doesn't detect my external monitor.

Reply:
What just came to my mind.
Are you guys using X11 or Wayland or wouldn't that make a difference?

Reply:
I use plasma with wayland

Reply:
Solved the suspend problem by removing the webcam option.
Sorry for the detour.